File: Dataset/Leetcode/train/35/186.java
 class Solution {
    public int XXX(int[] nums, int target) {
        if(nums == null || nums.length == 0) {
            return -1;
        }
        int l = 0, r = nums.length - 1, result = -1;
        while(l <= r) {
            int mid = l + (r - l) / 2;
            if(target == nums[mid]) {
                return mid;
            }
            else if(target < nums[mid]) {
                if(target >= nums[l]) {
                    r = mid - 1;
                }
                else {
                    return l;
                }
            }
            else if(target > nums[mid]) {
                if(target <= nums[r]) {
                    l = mid + 1;
                }
                else {
                    return r + 1;
                }
            }
        }
        return result;
    }
}

File: Dataset/Leetcode/train/13/186.py
class Solution:
    def XXX(self, s: str) -> int:
        l = ['I','V','X','L','C','D','M']
        l1 = [1,5,10,50,100,500,1000]
        h = {}
        for i in range(7):
            h[l[i]] = l1[i]
        r = 0
        for i in s:
            r += h[i]
        if 'IV' in s:
            r -= 2
        if 'IX' in s:
            r -= 2
        if 'XL' in s:
            r -= 20
        if 'XC' in s:
            r -= 20
        if 'CD' in s:
            r -= 200
        if 'CM' in s:
            r -= 200
        return r
            
            

